storybooks/storybook
# WIP: Support BigInt for controls

Closes https://github.com/storybookjs/storybook/issues/22452






## What I did

Providing support for bigint controls.

## Checklist for Contributors

### Testing



#### The changes in this PR are covered in the following automated tests:

#### Manual testing

1. Start Storybook Blocks
2. Go to http://localhost:6006/?path=/story/controls-object--object
3. Play around with the configured BigInt value. Editing should work without issues. The implementation has one small downside: The `n` suffix is not shown in the plaintext view of the Object Story. Also, editing in raw mode doesn't work. But at least Storybook does not crash anymore:
<img width="444" alt="Bildschirmfoto 2023-10-23 um 15 50 04" src="https://github.com/storybookjs/storybook/assets/5889929/90f79179-55c8-461c-a360-9c1828916a12">




### Documentation



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release



This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_
